include/arch/posix/arch.h

```diff
@@ -47,6 +48,10 @@ typedef struct __esf z_arch_esf_t;
 
 extern u32_t z_timer_cycle_get_32(void);
 
+void posix_halt_cpu(void);
```

Comment:
Nit: It is a bit ugly to redefine here these prototypes instead of pulling the posix_sox_if.h
I would not think it is much of a problem to have other users indirectly including that header, or? Meaning if need be we can consider it "public"

Reply:
Ok, I will change `posix_soc_if.h` to be public again.

I will have to do `#include <arch/posix/posix_soc_if.h>` though.

Reply:
That sounds ok. Thanks.

By the way, let me apologize for what seems like me changing my mind and having you wait for too long. I have been out of the office for 2 weeks in the last 3, and during a too quick review I probably did not fully understand where you were going with this PR.

Reply:
I moved `posix_soc_if.h` back to public (`include/arch/posix/`) and applied the same deprecation strategy used on `posix_trace.h` on it (I suppose since it is required by certain public APIs, you would want them implicitly included as well).